### PR TITLE
feat: replace shelljs.exec with empty function on git operation

### PR DESCRIPTION
**Pull request by slavik-lvovsky**

### CODE REVIEW

File: packages/backend/src/utils/shellJsHelper.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+export const applyExecWorkaround = function () {
+  const originalRequire = Module.prototype.require;
+  Module.prototype.require = function (request: string) {
+    if (request.includes("shelljs")) {
+      const shellJsModule = originalRequire.apply(this, arguments);
+      if (shellJsModule.exec) {
+        const originalExec = shellJsModule.exec;
+        shellJsModule.exec = function () {
+          if (arguments[0].startsWith("git config --get")) {
```
Comment (alex-gilin) [inline packages/backend/src/utils/shellJsHelper.ts:11]: what about command like "git (several spaces)   config \t --get" ?

**Review by alex-gilin — COMMENTED** [review]
I'm a little confused - what is this mock for? Git command doesn't crash, but it doesn't return the desired output - what is the scenario ?

File: packages/backend/src/utils/shellJsHelper.ts (hunk 1 above)
Comment (slavik-lvovsky) [reply packages/backend/src/utils/shellJsHelper.ts]: I have added an explanation why do we need this.
Added unit tests

**Review by alex-gilin — COMMENTED** [review]
https://github.com/shelljs/shelljs/wiki/Electron-compatibility --> whether the 'known workaround' they proposal there, doesn't work for us ?

**Comment by slavik-lvovsky** [comment]
> https://github.com/shelljs/shelljs/wiki/Electron-compatibility --> whether the 'known workaround' they proposal there, doesn't work for us ?

No it does not work

**Review by alex-gilin — APPROVED** [review]

**Pull request merged**